src/game/Spell.cpp

```diff
@@ -2982,9 +2982,12 @@ void Spell::cast(bool skipCheck)
     // CAST SPELL
     SendSpellCooldown();
 
-    TakePower();
-    TakeReagents();                                         // we must remove reagents before HandleEffects to allow place crafted item in same slot
-    TakeAmmo();
+    if (!m_IsTriggeredSpell)
```

Comment:
Oh sorry, I merged those two account. I don't think there was anything worth it on there because those fixes are acient and most of them not done correctly. I think there is a commit about Devastate on NostTBC which is way cleaner.